packages/shared/lib/core/wallet/utils/outputs/isOutputTimeLocked.ts

```diff
@@ -1,9 +1,16 @@
-import { getTimelockDateFromOutput } from '@core/wallet'
-import { CommonOutput, OutputData } from '@iota/sdk/out/types'
+import { CommonOutput, OutputData, TimelockUnlockCondition, UnlockConditionType } from '@iota/sdk/out/types'
+import { nodeInfoProtocolParameters } from '@core/network'
+import { get } from 'svelte/store'
 
 export function isOutputTimeLocked(outputData: OutputData): boolean {
-    const output = outputData.output
-    const unlockTime = getTimelockDateFromOutput(output as CommonOutput)
-
-    return unlockTime !== undefined && unlockTime.getTime() > Date.now()
+    const output = outputData.output as CommonOutput
+    const timelockUnlockCondition = output.unlockConditions.find(
+        (unlockCondition) => unlockCondition.type === UnlockConditionType.Timelock
+    ) as TimelockUnlockCondition
+    const nodeProtocolParameters = get(nodeInfoProtocolParameters)
+    if (!nodeProtocolParameters || !timelockUnlockCondition) {
+        return true
+    } else {
+        return timelockUnlockCondition.slotIndex + nodeProtocolParameters.minCommittableAge < 0
```

Comment:
I don't think this makes sense, this is not the same as the rust [code](https://github.com/iotaledger/iota-sdk/blob/2.0/sdk/src/types/block/output/unlock_condition/timelock.rs#L34).

The correct (I Think) formula is: `(currentSlotIndex + minCommittableAge) < output.timelockUnlockCondition.slotIndex`

but you are doing: `(output.timelockUnlockCondition.slotIndex + + minCommittableAge) < 0`

If you think about it, you can't check if an output is locked if you don't even know at what moment you are when you check it.

Reply:
From another POV, there is no way that your `(output.timelockUnlockCondition.slotIndex + minCommittableAge)` can ever be `< 0`

Reply:
So I think the correct code would be something like 

```ts
(currentSlotIndex + nodeProtocolParameters.minCommittableAge) < timelockUnlockCondition.slotIndex
```

I don't know where you can't get `currentSlotIndex` from tho

Reply:
Ah, I see there is `getSlotIndexFromNodeInfo` you can use that for `currentSlotIndex`